cmd/login/aws.go

```diff
@@ -0,0 +1,158 @@
+package login
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/k8sclient/v7/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	eks "sigs.k8s.io/cluster-api-provider-aws/controlplane/eks/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/giantswarm/kubectl-gs/v2/pkg/kubeconfig"
+)
+
+type eksClusterConfig struct {
+	clusterName          string
+	certCA               []byte
+	controlPlaneEndpoint string
+	filePath             string
+	loginOptions         LoginOptions
+	region               string
+
+	awsProfileName string
+}
+
+// storeWCClientCertCredentials saves the created client certificate credentials into the kubectl config.
+func storeWCAWSIAMKubeconfig(k8sConfigAccess clientcmd.ConfigAccess, c eksClusterConfig, mcContextName string) (string, bool, error) {
+	config, err := k8sConfigAccess.GetStartingConfig()
+	if err != nil {
+		return "", false, microerror.Mask(err)
+	}
+
+	if mcContextName == "" {
+		mcContextName = config.CurrentContext
+	}
+	contextName := kubeconfig.GenerateWCAWSIAMKubeContextName(mcContextName, c.clusterName)
+	userName := fmt.Sprintf("%s-user", contextName)
+	clusterName := contextName
+
+	contextExists := false
+
+	{
+		// Create authenticated user.
+		user, exists := config.AuthInfos[userName]
+		if !exists {
+			user = clientcmdapi.NewAuthInfo()
+		}
+
+		user.Exec = &clientcmdapi.ExecConfig{
+			APIVersion: "client.authentication.k8s.io/v1beta1",
+			Command:    "aws",
+			Args:       []string{"--region", c.region, "eks", "get-token", "--cluster-name", c.clusterName, "--output", "json"},
```

Comment:
this is copied from how it looks like when you create kubeconfig via `aws eks update-kubeconfig`